### ISSUE TITLE
Type checking support 

### ISSUE THREAD

**Issue by eliotwrobson** [martenlienen/cached_method#1]
It would be great if this decorator preserved type annotations and had type checking support.

**Comment by martenlienen** [comment]
I agree. Do you have a suggestion how to implement it with the `cached_method` class? The quickest way to a solution would be for you to send a PR.

**Comment by eliotwrobson** [comment]
It should be possible to do something similar to the annotations used here: https://github.com/penguinolog/backports.cached_property/blob/master/backports/cached_property/__init__.pyi

However, I'm not quite sure how to test this locally.

**Comment by martenlienen** [comment]
I don't know if it is possible because [`lru_cache`](https://github.com/python/cpython/blob/main/Lib/functools.py#L479) also does not have type annotations.

**Comment by eliotwrobson** [comment]
I don't think the lack of annotations on lru cache should be an issue. I think we can declare annotations here that will still get used.

**Comment by martenlienen** [comment]
Yes, that is true. Unfortunately, I don't have the capacity to do so at the moment, though if you want to see that change, I would appreciate an implementation.
